### PR TITLE
created quoteToInvoice converter api

### LINKED ISSUE #261: Create Invoice from Quotation API
Create new Quote api that can convert Quote to invoice :

From Quote Data , Create new invoice in Invoice table

### PR DESCRIPTION
**Pull request by Elham-EN**
This change resolve #261 by creating the Quote to Invoice converter api.

### CODE REVIEW

**Comment by salahlalami** [comment]
@Elham-EN Thanks ,
please add this method to quote route api

**Comment by salahlalami** [comment]
@onfranciis please test it and try to enhance

File: controllers/erpControllers/quoteController.js [hunk 1]
```diff
@@ -140,6 +140,78 @@ methods.update = async (req, res) => {
...
+
+    // Mark the quote as converted
+    quote.converted = true;
+    await quote.save();
+
+    // Return the created invoice
+    return res.status(200).json({
+      success: true,
+      result: invoice,
+      message: 'Successfully converted quote to invoice',
```
Comment (salahlalami) [inline controllers/erpControllers/quoteController.js:202]: return quote as result instead.

**Review by salahlalami — CHANGES_REQUESTED** [review]

File: controllers/erpControllers/quoteController.js [hunk 2]
```diff
@@ -140,6 +140,78 @@ methods.update = async (req, res) => {
...
+      return res.status(404).json({
+        success: false,
+        result: null,
+        message: 'Quote not found',
+      });
+    }
+
+    // If the quote is already converted, prevent creating another invoice
+    if (quote.converted) {
+      return res.status(400).json({
```
Comment (onfranciis) [inline controllers/erpControllers/quoteController.js:161]: Kindly use a status 409 to indicate the conflict error

**Comment by onfranciis** [comment]
The updated pull request is now here #327

**Pull request closed without merging**